packages/cli/generators/relation/index.js

```diff
@@ -338,50 +430,64 @@ module.exports = class RelationGenerator extends ArtifactGenerator {
 
   async promptRelationName() {
     if (this.shouldExit()) return false;
-    if (
-      this.artifactInfo.relationType === relationUtils.relationType.belongsTo
-    ) {
-      this.artifactInfo.destinationModelPrimaryKey = await relationUtils.getModelPrimaryKeyProperty(
-        this.fs,
-        this.artifactInfo.modelDir,
-        this.artifactInfo.destinationModel,
-      );
-      if (this.artifactInfo.destinationModelPrimaryKey == null) {
-        return this.exit(
-          new Error(ERROR_DESTINATION_MODEL_PRIMARY_KEY_DOES_NOT_EXIST),
-        );
-      }
-
-      this.artifactInfo.destinationModelPrimaryKeyType = relationUtils.getModelPropertyType(
-        this.artifactInfo.modelDir,
-        this.artifactInfo.destinationModel,
-        this.artifactInfo.destinationModelPrimaryKey,
-      );
-    }
-
     if (this.options.relationName) {
       debug(
         `Relation name received from command line: ${this.options.relationName}`,
       );
       this.artifactInfo.relationName = this.options.relationName;
     }
+    this.artifactInfo.defaultRelationName = this._getDefaultRelationName();
+    // for hasMany && hasOne, the source key is the same as the relation name
+    const msg =
+      this.artifactInfo.relationType === 'belongsTo'
+        ? PROMPT_MESSAGE_RELATION_NAME
+        : PROMPT_MESSAGE_PROPERTY_NAME;
 
     return this.prompt([
       {
         type: 'string',
         name: 'relationName',
-        message: PROMPT_MESSAGE_PROPERTY_NAME,
+        message: msg,
         when: this.artifactInfo.relationName === undefined,
-        default: this._getDefaultRelationName(),
+        default: this.artifactInfo.defaultRelationName,
       },
     ]).then(props => {
       debug(`props after relation name prompt: ${inspect(props)}`);
+      // checks if the relation name already exist
+      this.artifactInfo.srcRepositoryFile = path.resolve(
+        this.artifactInfo.repoDir,
+        utils.getRepositoryFileName(this.artifactInfo.sourceModel),
+      );
+      this.artifactInfo.srcRepositoryClassName =
+        utils.toClassName(this.artifactInfo.sourceModel) + 'Repository';
+      this.artifactInfo.srcRepositoryFileObj = new relationUtils.AstLoopBackProject().addExistingSourceFile(
+        this.artifactInfo.srcRepositoryFile,
+      );
+
+      const repoClassDeclaration = this.artifactInfo.srcRepositoryFileObj.getClassOrThrow(
+        this.artifactInfo.srcRepositoryClassName,
+      );
+      // checks if the relation name already exists in repo
+      if (
+        relationUtils.doesPropertyExist(
```

Comment:
Just curious: I think the relation definition is in the model file, why do we have to use more steps to get the corresponding repository file and check the existence?

Reply:
Good question! From my understanding the model has _the property_ (e.g `orders` or `customerId`). And the repository has _the relation name_ (e.g `orders` or `customer`). They might have the same name but most of time they don't. For example, for belongsTo relation, the relation name is `customer`.

The generator throws errors if the repository already has the relation name defined. And the generator didn't have such checks before .

Reply:
>  And the repository has the relation name

Oh that's smart! 👍 makes perfect sense then